### DISCUSSION TITLE
RuntimeError: Cannot re-initialize CUDA in forked subprocess. To use CUDA with multiprocessing, you must use the 'spawn' start method

### DISCUSSION THREAD

**Discussion by sapkun** [huggingface/diffusers#6985] [General]
Hello, when i train my own data with the sript [train_instruct_pix2pix_sdxl.py](https://github.com/huggingface/diffusers/blob/main/examples/instruct_pix2pix/train_instruct_pix2pix_sdxl.py) with `--dataloader_num_workers=2`, it gives me this error: 
`RuntimeError: Cannot re-initialize CUDA in forked subprocess. To use CUDA with multiprocessing, you must use the 'spawn' start method`. 
But If I change the `--dataloader_num_workers=0`, it will work. How to run this script with multiple `dataloader_num_workers`?
Can anyone help in this regard? thank you!

**Comment by sayakpaul** [comment]
You can refer to https://github.com/sayakpaul/instructpix2pix-sdxl for a more dedicated treatment of that script to run on multiple GPUs.
